### DISCUSSION TITLE
Any chat solution at all

### DISCUSSION THREAD

**Discussion by nacho-domatix** [tteck/Proxmox#567] [Ideas]
I've tried already Mattermost and using Rocket.chat until now. Both are correct, but lack of 'ambition' and community leadership (they have enterprise version and don't support video calls).

Since "matrix" based solutions can be kind of hard to maintain (I like so much Element cause its features), I'd go for something simpler. Not sure what's the best solution at all, but since I love Rust & Go for this kind of solutions I'd suggest:

https://gotify.net
https://github.com/revoltchat

None of them have videocalls (yet) anyway. But they look promising.

I'd love to read other people's suggestion anyway. And looking forward to try a new chat solution.

BTW Nextcloud support chat + videocalls + mobile apps (tho is not an 'exclusive' chat solution)
Jitsi is a great software too, but just provides chats when you're in a videocall.

**Comment by MrEddX** [comment]
I personally use [Prosody ](https://prosody.im/) and [Ejabberd](https://www.ejabberd.im/). Great solutions, relatively easy to configure when you get used to.

- Prosody's config file is more readable since it uses the Lua syntax.
- With Ejabberd you get most configs done out of the box, and has Web based administration.
- The most time consuming part is: Firewall and DNS configuration since they use many ports.
- In terms of clients: Desktop, Web, Mobile (Android and iOS).
- Both support: Chat, Voice, Video calls.
- Both are running super stable on low powered servers (1 x ARM CPU and 1GB RAM). I've never seen RAM consumption above 300MB with 20-30 clients. 
- Missing things: Stickers (If they are important to you, check [Matrix](https://matrix.org/)).
- Good tutorials: [Prosody](https://jalogisch.de/2018/modern-selfhosted-xmpp-jabber/), [Ejabberd](https://www.process-one.net/blog/how-to-move-the-office-to-real-time-im-on-ejabberd/), [Matrix](https://github.com/spantaleev/matrix-docker-ansible-deploy).

BTW Gotify is great if you need server notifications/reports. I think this is one of the easiest setups for self-hosted push notification services. It is integrated in [Acme.sh](https://github.com/acmesh-official/acme.sh/blob/master/notify/gotify.sh), [UptimeKuma](https://github.com/louislam/uptime-kuma/blob/master/src/components/notifications/Gotify.vue) an so on. But since it has also a [CLI ](https://www.ejabberd.im/) client, that means you have no limitations. Just love the simplicity of this project.

**Reply by nacho-domatix** [reply]
I found Prosody too basic. Wasn't unable to try Ejabberd as I didn't see any demo.

**Comment by MrEddX** [comment]
Currently I'm working on a test Ejabberd implementation. So if you like I can give you a demo account(s) to play with it. Just let me know where to send the credentials.

**Reply by nacho-domatix** [reply]
Can you post it here? So anyone can see it and give an opinion.

**Reply by MrEddX** [reply]
I don't think it's a good idea to make the access data public. The service does not have a web-based interface and is only accessible through a [client](https://xmpp.org/software/clients/).

**Reply by nacho-domatix** [reply]
Oh, I see. If it doesn't have a web client it won't fullfit my requirements.  Doesn't mean is not valid, but what I'm seeking is an "all-in-one" solution for PCs who does not have permissions to install anything.

**Reply by MrEddX** [reply]
Not on this server, but there is a public web client available [Here](https://conversejs.org/fullscreen.html), which can be used with any server. We have just decided that we don't need one on the current server. The integration is just a few lines in the ejabberd config file.
